pkg/driver/trace_driver.go

```diff
@@ -100,13 +103,235 @@ func (d *Driver) runCSVWriter(records chan interface{}, filename string, writerD
 	writerDone.Done()
 }
 
-func DAGCreation(functions []*common.Function) *list.List {
+func createDAGWorkflow(functionList []*common.Function, function *common.Function, maxWidth int, maxDepth int) *list.List {
+	// Generating the number of Invocations per Depth
+	widthList := []int{}
+	widthList = append(widthList, 1)
+	for i := 1; i < maxDepth-1; i++ {
+		widthList = append(widthList, (rand.Intn(maxWidth-widthList[i-1]+1) + widthList[i-1]))
+	}
+	widthList = append(widthList, maxWidth)
+	queue := []*list.Element{}
 	linkedList := list.New()
-	// Assigning nodes one after another
-	for _, function := range functions {
-		linkedList.PushBack(function)
+	functionIndex := []int{}
+	for i := range functionList {
+		functionIndex = append(functionIndex, getName(functionList[i]))
+	}
+	functionId := getName(function)
+	functionIndex = append(functionIndex[:functionId], functionIndex[functionId+1:]...)
+	var totalFunctionsToInvoke int64
+	var additionalBranches int
+	var listElement *list.Element
+	var child *common.Node
+	// Using -1 as indicator for dummy node
+	dummyNode := &common.Node{Depth: -1}
+	node := &common.Node{Function: function, Depth: 0}
+	totalFunctionsToInvoke += 1
+	initialList := linkedList
+	// First sequential invocation
+	if maxDepth == 1 {
+		linkedList.PushBack(dummyNode)
+		linkedList.Front().Value = node
+		return initialList
```

Comment:
May I kindly ask why the function is returning initialList here? Based on lines 115 and 130, it appears that directly returning initialList might result in returning an empty list.

Reply:
Thanks! Looking back, the initialList is referring to linkedList, thus it is irrelevant. Will change it accordingly.